`jmwalletd` exits after failing to start a maker via rpc (in `regtest`)

After this error described below, `jmwalletd` exists. This happened on regtest.

Steps to reproduce:
- Have one single expired fidelity bond utxo
- Start the maker via call to `POST /api/v1/wallet/$wallet_name/maker/start`

This reminds me of and is similar to https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/1213.
But in this case, it is really a nasty edge case, so probably rather low priority.

This is the full log:
```
2022-07-05 11:39:51,573 [INFO]  JoinMarket 0.9.7dev
2022-07-05 11:39:51,580 [DEBUG]  connection was made, starting client.
2022-07-05 11:39:51,582 [INFO]  Attempting to start onion service on port: 5222 ...
2022-07-05 11:39:51,588 [DEBUG]  building irc
2022-07-05 11:39:51,588 [DEBUG]  build_irc:  irc 6667 #joinmarket-pit-test
2022-07-05 11:39:54,736 [INFO]  Timelocked UTXO at 'm/84'/1'/0'/2/22:1635724800' has been auto-frozen. They cannot be spent by makers.
2022-07-05 11:39:54,736 [ERROR]  do not have any coins left
2022-07-05 11:39:54,737 [DEBUG]  rpc: getblockchaininfo []
2022-07-05 11:39:54,742 [DEBUG]  rpc: getblockheader ['3df219f03fd004f3a5b0d9be878fe3b7ea28f830d574f21786992a31505f2501']
2022-07-05 11:39:54,744 [INFO]  Announcing fidelity bond coin e5a9a4974ad48b226014497f0d2bd491adf276c66dd648c1a71ae3cffb82512b:0
2022-07-05 11:39:54,744 [ERROR]  Failed to create offers.
2022-07-05 11:39:54,761 [DEBUG]  signedOn:  ergo.test
2022-07-05 11:39:55,074 [INFO]  Shutting down YieldGenerator service.
2022-07-05 11:39:56,024 [INFO]  joined: #joinmarket-pit-test ergo.test
2022-07-05 11:39:56,024 [INFO]  Could not connect to *ALL* servers yet, waiting up to 60 more seconds.
2022-07-05 11:50:54,029 [INFO]  Failed to start after 10 minutes, giving up.
2022-07-05 11:50:54,031 [INFO]  Lost IRC connection to: ergo.test . Should reconnect automatically soon.
2022-07-05 11:50:54,033 [DEBUG]  IRC connection lost: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionLost'>: Connection to the other side was lost in a non-clean fashion: Connection lost.
]
```

After that last log message, `jmwalletd` exits.
```
jm_regtest_joinmarket        | 2022-07-05 11:04:00,466 INFO exited: jmwalletd (exit status 0; expected)
```

The odd thing here is... `e5a9a4974ad48b226014497f0d2bd491adf276c66dd648c1a71ae3cffb82512b:0` (`m/84'/1'/0'/2/22:1635724800`) **is** the expired fidelity bond..
It says: 
```
2022-07-05 11:39:54,736 [INFO]  Timelocked UTXO at 'm/84'/1'/0'/2/22:1635724800' has been auto-frozen. They cannot be spent by makers.
```
*Side note*: After restarting the daemon, the utxo is **not** frozen.


But then after that, the FB is announced (or at least this message is logged):
```
2022-07-05 11:39:54,744 [INFO]  Announcing fidelity bond coin e5a9a4974ad48b226014497f0d2bd491adf276c66dd648c1a71ae3cffb82512b:0
```

This might really be a regtest issue (just speculating), however, the FB has been created with a lockdate in the past and was **not** auto-frozen automatically contrary to coins of "normal" FBs (locktime in the future). Regardless of this, `jmwalletd` should probably not exit. What do you think? 


This error seems to be correct:
```
2022-07-05 11:39:54,736 [ERROR]  do not have any coins left
```

However, a call to `POST /marker/start` returns `202 Accepted`, whereas when it is invoked with all utxos frozen it will respond with `409 Conflict` and body `{ "message": "Maker could not start, no confirmed coins." }` (as done as a fix for https://github.com/JoinMarket-Org/joinmarket-clientserver/issues/1213).

OK I was able to reproduce this on regtest, just using CLI yg, not RPC. And in so doing remembered some details:

(deleted a couple of irrelevant lines):

```
2022-07-14 21:38:19,942 [INFO]  JoinMarket 0.9.7dev
2022-07-14 21:38:19,942 [INFO]  starting yield generator
2022-07-14 21:38:19+0100 [-] Log opened.
2022-07-14 21:38:19,944 [INFO]  Joinmarket daemon listening on port 27186
2022-07-14 21:38:19,944 [INFO]  Starting transaction monitor in walletservice
2022-07-14 21:38:19,948 [DEBUG]  connection was made, starting client.
2022-07-14 21:38:19,951 [DEBUG]  building irc
2022-07-14 21:38:19,951 [DEBUG]  build_irc:  localhost 16667 #joinmarket-pit-test
2022-07-14 21:38:19,951 [DEBUG]  building irc
2022-07-14 21:38:19,952 [DEBUG]  build_irc:  localhost 16668 #joinmarket-pit-test
2022-07-14 21:38:21,963 [INFO]  Timelocked UTXO at 'm/84'/1'/0'/2/21:1633046400' has been auto-frozen. They cannot be spent by makers.
2022-07-14 21:38:21,963 [ERROR]  You do not have the minimum required amount of coins to be a maker: 100000
Try setting txfee_contribution to zero and/or lowering the minsize.
2022-07-14 21:38:21,964 [DEBUG]  rpc: getblockchaininfo []
2022-07-14 21:38:21,968 [DEBUG]  rpc: getblockheader ['64e03182213efecbb8102b517bcbbf99f151e4ba59709dcd29414ca14f47de27']
2022-07-14 21:38:21,970 [INFO]  Announcing fidelity bond coin c5f3c0b1276bb5a83556e98fb9c46bd9291bea44cbef8980c51f43651270f071:0
2022-07-14 21:38:21,970 [ERROR]  Failed to create offers.
```

First point: "Announcing fidelity bond" is *not* an error here. Fidelity bonds *do* have value after expiration, surprisingly. See the detailed reasoning in @chris-belcher 's writeup: https://gist.github.com/chris-belcher/87ebbcbb639686057a389acb9ab3e25b .

Second: "Timelocked UTXO has been auto-frozen" is also not an error. This is one of those details I forget after a while, but it's easy to remember the reasoning: we auto-freeze TL utxos for the maker whether they are unlocked or not, as it's the easiest way to handle the issue that we don't want to use them as maker, in coinjoins.

As you can see the second one is a bit of an issue for the RPC as distinct from the usual CLI way of running the maker. Because if the maker service is stopped, but the jmwalletd is still running of course, the utxo will stay frozen/locked. I might argue (I think I will!) that this is not a huge deal. These utxos have to be spent manually anyway. So just unfreeze them when you need to.

What's left here is as you correctly point out, we have a scenario similar to #1213 : we don't want the maker starting and hanging here, we want it to just refuse to start, since it doesn't actually have offer-able coins. So a fix similar to #1242 is in order. The only problem is that we now want "get balance not including timelocked coins, even if they are unlocked" and the code in `jmclient.wallet` doesn't actually support that yet, I believe. I'll try to do that.

--- a/jmclient/jmclient/wallet_rpc.py
+++ b/jmclient/jmclient/wallet_rpc.py
@@ -25,7 +25,7 @@ from jmclient import Taker, jm_single, \
     NotEnoughFundsException, get_tumble_log, get_tumble_schedule, \
     get_schedule, get_tumbler_parser, schedule_to_text, \
     tumbler_filter_orders_callback, tumbler_taker_finished_update, \
-    validate_address
+    validate_address, FidelityBondMixin
 from jmbase.support import get_log, utxostr_to_utxo
 
 jlog = get_log()
@@ -701,6 +701,17 @@ class JMWalletDaemon(Service):
                     # note: this raise will prevent the setup
                     # of the service (and therefore the startup) from
                     # proceeding:
+                    raise NotEnoughCoinsForMaker()
+                # We must also not start if the only coins available are of
+                # the TL type *even* if the TL is expired. This check is done
+                # here early, as above, to avoid the maker service starting.
+                utxos = self.services["wallet"].get_all_utxos()
+                # remove any TL type:
+                utxos = [u for u in utxos.values() if not \
+                         FidelityBondMixin.is_timelocked_path(u["path"])]
+                # Note that only the following check is required since we
+                # already checked that balance is non-zero.
+                if len(utxos) == 0:
                     raise NotEnoughCoinsForMaker()
 
             self.services["maker"].addCleanup(cleanup)

